### PR TITLE
add quoting to postgis raster driver

### PR DESCRIPTION
**Pull request by jef-n**
## What does this PR do?
add quoting to the postgisraster driver - untested, here to trigger tests…
## What are related issues/pull requests?
https://issues.qgis.org/issues/19583
## Tasklist

 - [ ] ADD YOUR TASKS HERE
 - [ ] Add test case(s)
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

## Environment

Provide environment details, if relevant:

* OS:
* Compiler:

### CODE REVIEW

File: autotest/ogr/ogr_nas.py [hunk 1]
```diff
@@ -342,9 +342,6 @@ def ogr_nas_5():
     ogr_nas_4,
     ogr_nas_5]
 
-# NAS tests disabled for now
-gdaltest_list = []
```
Comment (rouault) [inline autotest/ogr/ogr_nas.py]: this change is unrelated to the commit, and will break tests

File: gdal/ogr/ogrsf_frmts/nas/ogrnasdriver.cpp [hunk 2]
```diff
@@ -74,7 +74,11 @@ static int OGRNASDriverIdentify( GDALOpenInfo* poOpenInfo )
     if( strstr(szPtr,"opengis.net/gml") == nullptr )
         return FALSE;
 
-    char **papszIndicators = CSLTokenizeStringComplex(CPLGetConfigOption("NAS_INDICATOR", ""), ";", 0, 0 );
+    char **papszIndicators = CSLTokenizeStringComplex(
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/nas/ogrnasdriver.cpp]: unrelated change

File: gdal/ogr/ogrsf_frmts/nas/drv_nas.html [hunk 3]
```diff
@@ -15,11 +15,9 @@ <h1>NAS - ALKIS</h1>
 library.</p>
 
 <p>The driver looks for "opengis.net/gml" and one of the strings semicolon
-separated strings listed in the option <b>NAS_INDICATOR</b>  to determine if
-a input is a NAS file and ignores all files without any matches.</p>
-
-<p>The former default of "NAS-Operationen;AAA-Fachschema;aaa.xsd;aaa-suite" was
-removed and has now to be explicitly specified to enable detection.</p>
+separated strings listed in the option <b>NAS_INDICATOR</b> (which defaults to
```
Comment (rouault) [inline gdal/ogr/ogrsf_frmts/nas/drv_nas.html]: unrelated change

**Comment by rouault** [comment]
oh, looking closer, I didn't see there was a separate commit for the change in NAS, so this is probably OK (although a bit confusing to be conflated in this PR)

**Comment by jef-n** [comment]
Yes sorry.  The NAS change is just one pending in my branch, that just reverts the recent changes back to how it was before (OGR_SKIP=NAS should be ok after all).    With the PR I actually only wanted to trigger test - which miserably failed - although the failures seem unrelated to the changes.

**Comment by rouault** [comment]
I've restarted the failed jobs

File: gdal/frmts/postgisraster/postgisrasterdataset.cpp [hunk 4]
```diff
@@ -2104,12 +2120,12 @@ const char * pszValidConnectionString)
             papszSubdatasets[2 * i] =
                 CPLStrdup(CPLSPrintf("SUBDATASET_%d_NAME=PG:%s schema=%s table=%s column=%s "
                     "where='%s = %s'", i+ 1, pszValidConnectionString,
-                    pszSchema, pszTable, pszColumn, pszPrimaryKeyName,
+                    pszSchema, pszTable, pszColumn, osPrimaryKeyNameI.c_str(),
```
[the remark this answers is not in the document]
Comment (jef-n) [reply gdal/frmts/postgisraster/postgisrasterdataset.cpp]: Should work - Looks like what helped in qgis/QGIS@f4d08eb926

**Comment by rouault** [comment]
Looks good to me. I let you merge
